lib/ansible/module_utils/urls.py

```diff
@@ -1479,7 +1479,8 @@ def fetch_url(module, url, data=None, headers=None, method=None,
     client_cert = module.params.get('client_cert')
     client_key = module.params.get('client_key')
 
-    cookies = cookiejar.LWPCookieJar()
+    if not isinstance(cookies, cookiejar.CookieJar):
```

Comment:
This should probably be
``` python
if cookies is None:
```

If cookies was neither None nor a cookiejar.CookieJar, a TypeError or similar should be raised (although the idiom in python is not to check for that).  The present code could mask coding errors (Say I pass in a list of cookies instead of a cookie jar.... I might think the code will do the right thing because I don't get an error but in reality, this code will silently discard my list of cookies and use a new, empty cookiejar instead.

This is not a blocker for merging the backport but it should be changed in devel and perhaps backported as well..